hbase-common/src/main/java/org/apache/hadoop/hbase/util/UnsafeAvailChecker.java

```diff
@@ -168,20 +174,24 @@ public Boolean run() {
     }
   }
 
+  public static void overrideUnsafeAvail(boolean unsafeAvail) {
+    useUnsafe = unsafeAvail;
```

Comment:
If this is changed by the application it will trigger de-opt and JIT method replacement at the converter call sites, but the performance blip will be temporary, because the new case will also predict the same return in 100% of cases. (Unless the application keeps flipping it, but that is nonsensical, we don't need to consider it.)